service/src/main/java/com/google/auto/service/processor/AutoServiceProcessor.java

```diff
@@ -127,26 +131,32 @@ private void processAnnotations(Set<? extends TypeElement> annotations,
     for (Element e : elements) {
       // TODO(gak): check for error trees?
       TypeElement providerImplementer = (TypeElement) e;
-      AnnotationMirror providerAnnotation = getAnnotationMirror(e, AutoService.class).get();
-      DeclaredType providerInterface = getProviderInterface(providerAnnotation);
-      TypeElement providerType = (TypeElement) providerInterface.asElement();
-
-      log("provider interface: " + providerType.getQualifiedName());
-      log("provider implementer: " + providerImplementer.getQualifiedName());
-
-      if (!checkImplementer(providerImplementer, providerType)) {
-        String message = "ServiceProviders must implement their service provider interface. "
-            + providerImplementer.getQualifiedName() + " does not implement "
-            + providerType.getQualifiedName();
-        error(message, e, providerAnnotation);
-      }
+      AnnotationMirror annotationMirror = getAnnotationMirror(e, AutoService.class)
+          .get(); // We already know it's present
+      Set<DeclaredType> providerInterfaces = getFieldOfClasses(
+          annotationMirror,
+          "value",
+          processingEnv.getElementUtils());
+      for (DeclaredType providerInterface : providerInterfaces) {
+        TypeElement providerType = (TypeElement) providerInterface.asElement();
```

Comment:
I know this is a move, but might as well do some tiding up while we're here:
`MoreTypes.asTypeElement(providerInterface)`

Reply:
7578b71 (and updated for the new API in dfbbdd4 after updating guava)